src/Credentials/ExternalAccountCredentials.php

```diff
@@ -100,6 +100,9 @@ public function __construct(
 
         if (array_key_exists('service_account_impersonation_url', $jsonKey)) {
             $this->serviceAccountImpersonationUrl = $jsonKey['service_account_impersonation_url'];
+        } else {
+            //If we do not initalize this, getCacheKey throws an error.
+            $this->serviceAccountImpersonationUrl = null;
         }
```

Comment:
more concise!
```suggestion
        $this->serviceAccountImpersonationUrl = $jsonKey['service_account_impersonation_url'] ?? null
```

Reply:
I have never been fan of this syntax haha. But is just an opinionated matter. Let me add it.

Reply:
Going from 5 lines to 1 is more readable IMHO